### DISCUSSION TITLE
[Feedback] Need more recent snapshots

### DISCUSSION THREAD

**Discussion by jchung00** [ethereum-optimism/developers#44] [Feedback]
### Issue Description

Saw this here: https://github.com/ethereum-optimism/optimism/issues/8027

It is almost impossible to sync a new Optimism node. Been trying for a week now and still not caught up. Server specs are much higher than what would normally be required.

Am hearing from other devs it took them almost a month to sync from the snapshots available.

Is there anywhere to get a more recent snapshot?

**Comment by sbvegan** [comment]
Hey @jchung00! Thank you for highlighting this pain point. At the moment, we do not have a more recent snapshot, but this has sparked some internal discussion on how we should approach this. 

One thing you can try to speed up your syncing process is specifying `l1.rpckind` on `op-node`. The fetching of receipts is a bottleneck and using this flag will allow the node to fetch them optimally.

From the `op-node --help`:

```
--l1.rpckind value                  (default: standard)                ($OP_NODE_L1_RPC_KIND)
          The kind of RPC provider, used to inform optimal transactions receipts fetching,
          and thus reduce costs. Valid options: alchemy, quicknode, infura, parity,
          nethermind, debug_geth, erigon, basic, any, standard
```

**Reply by moorow** [reply]
Hey @sbvegan!

If I am running my own L1 node with debug API enabled, is debug_geth the best setting to sync as quickly as possible?

Also are there any Cache settings that can be optimised to improve sync speed?

**Reply by sbvegan** [reply]
Hi @moorow, yep thats the option you want! If your machine has extra memory, you can bump up your `--cache`. However, I was recently reading this [geth blog post](https://blog.ethereum.org/2023/09/12/geth-v1-13-0) for v1.13.0 and they mentioned:

> No, we didn't drop the --cache flag, but chances are, you should!
> Geth's --cache flag has a bit of a murky past, going from a simple (and ineffective) parameter to a very complex beast, where it's behavior is fairly hard to convey and also to properly account.

The context is that geth is updating their database and `op-geth` is likely to follow. The upstream changes are already in the codebase, see the following [release](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101304.0). I'll look out for updated guidance from the client engineers.

**Reply by breezytm** [reply]
How much of an improve is expected using the `--l1.rpckind` flag? I haven't noticed a significant difference. The expected sync time is 16-20 days.

**Reply by sbvegan** [reply]
@breezytm that sync time sounds about right. I don't know the expected time improvement. What will happen is that your node will fetch receipts more efficiently and won't waste cycles on unsupported receipt fetching.

**Comment by sbvegan** [comment]
@jchung00 if this question has been answered for you, can you please mark it. Alternatively, maybe we can change the category to `Feedback` and others can signal their support for this here. What do you think about that?

**Reply by sbvegan** [reply]
I'm going to go ahead and make this a feedback thread to see if folks are willing to pay for this service.

**Reply by jchung00** [reply]
Sorry just saw this! Yes that's fine

**Comment by sbvegan** [comment]
Hey y'all, I have an update from our team. Internally we run archive nodes so we can create archive data directory snapshots to get node operators up and running faster. If there is an interest for this we can set it up where we post periodic up to date snapshots on our Google Cloud Provider box. This would allow anyone to download the snapshots. ~~This would require users to setup their own GCP accounts so they can pay for the downloads. Downloads of this size can cost a few hundred USD, so this is something we cannot provide for free.~~ Please note that we would not be making any profit, this would be a connivence provided for folks who want it. If this is something you want, please signal your support in this thread.

**Reply by moorow** [reply]
Awesome. thanks for looking into this! 

I only run full nodes but I would defs be willing to pay for snapshots.... I know a lot of people who would.

For anyone doing something commercial with a node, the ability to quickly download a snapshot and only have to sync a day / week is invaluable. 

It also would save a lot of people money and time from having to do backups of their data dirs....

Would be overall much more efficient for a 3rd party to provide frequent snapshots for a fee.

**Reply by smartcontracts** [reply]
As a quick point of clarification, it's a few hundred to do the upload and pretty cheap for subsequent downloads so it seems reasonable to do an upload 1-2 times per month until snap sync is ready. Would that be sufficiently recent to avoid these issues?

**Reply by sbvegan** [reply]
@moorow do you think you could have those folks signal their support here?

**Reply by sbvegan** [reply]
@smartcontracts thanks for the clarification, it looks like we have a couple options. So it would be helpful to know what frequency of snapshots users would want. Relaying what we've discussed internally, our options are:

- Upload once or twice a month where we eat the upload cost, but the download is free for users
- More frequent uploads and the user pays the convenience  premium on their own GCP account (~$300 download)

Important note, we're talking about archive node snapshots

**Reply by ghost** [reply]
Would be great if you could seed this though torrent which would save some traffic fees. And I think it would be good to distribute two different node snapshots, one for archive and one for pruned with path based state scheme

**Reply by web-dev100** [reply]
agreed with @kaliubuntu0206, differentiating between regular and archive would be a large improvement over the current (non existent) setup. a snapshot every week or so is more then sufficient for someone to catch up, for example this is the cadence of polygon's snapshots and that seems to work fine for anyone running a node on that chain. not having snapshots readily available will absolutely hurt Optimism's adoption and is in my opinion a major issue if not addressed.

**Reply by jchung00** [reply]
Even once or twice a month is fine. The current most recent public snapshot available is 4 months behind, which makes it impossible to sync an Optimism node in a reasonable timeframe. This would be much appreciated.

**Reply by sbvegan** [reply]
This feedback is great, thank you all!

**Reply by jchung00** [reply]
To add clarity, archive is not even needed for most users. Syncing a regular full node is already a massive pain point, and having a snapshot for a simple full node would be very helpful.

**Reply by imtipi** [reply]
i have a seedbox and i want to seed for full node snapshot torrent

**Reply by sbvegan** [reply]
Okay y'all, I'm following up with our developer infrastructure team today. I'll let you know what we land on.

**Comment by sbvegan** [comment]
Update: we have the team committed to uploading an archive node snapshot once a month. The team is working on putting together a project plan on how to implement this on the backend. I'll post more updates to this thread when I have them.

**Reply by sbvegan** [reply]
Update: the developer infrastructure team is scoping out the work and this likely to land in Q1 of 2024.

**Comment by onebyonee** [comment]
newer snapshot is really needed. with high end hardware u still need 2+ weeks to sync (in august) now its probably 3-4 weeks which is way too much for a l2 like opti

**Comment by ynvv** [comment]
Hello! We have actual snaps for OP node. More info in [this pull request](https://github.com/ethereum-optimism/docs/pull/355)

**Reply by sbvegan** [reply]
Hey @ynvv thank you again for providing this snapshot. For the folks using this snapshot, at the moment, when its extracted it has a top level directory with the `geth` directory inside. Context in this [comment](https://github.com/ethereum-optimism/docs/pull/355#issuecomment-1875786643).

**Comment by sbvegan** [comment]
Now we have archive snapshots as well: https://docs.optimism.io/builders/node-operators/management/snapshots

**Reply by tmeinlschmidt** [reply]
I started to download on 1/11 as upgrade didn't go well and still downloading... max speed 10MiB for downloading 3TB file is crazy :/ Even if you recommend using aria2c to 'speed up' downloading

**Reply by sbvegan** [reply]
Yeah we're seeing a lot of traffic, we've opened up more bandwidth and I've added a torrent [link](https://github.com/ethereum-optimism/docs/pull/429)

**Comment by ynvv** [comment]
Hello, guys!
We're just uploaded new snapshot with latest block [115039006](https://optimistic.etherscan.io/block/115039006) (after network update - Jan-19-2024 03:06:29 PM +UTC) at our site [datapoint.fastnode.io](https://datapoint.fastnode.io)

We have changed the structure of the archive: now only the geth folder is inside the archive (without the op-volume folder) (@sbvegan)

We also added the ability to download the first up-to-date snapshot of all that we have (details on the website above)

**Reply by sbvegan** [reply]
Incredible, thank you so much @ynvv!

**Comment by sbvegan** [comment]
I'm closing this out, our documentation now reflects up to date snapshots that node operators can use. Please note, when snap sync is enabled on OP Mainnet, the need for full node snapshots will go away because you'll be able to snap sync from peers :)
